### PR TITLE
network: also read mtime of drop-in configs

### LINKED ISSUE #15521: networkctl reload doesn't reload drop-in files unless the main file has a changed mtime
with the main config file in /etc/systemd/network, and a drop-in in /run/systemd/network/wg0.network.d, the changes in the drop-in are not applied unless `touch /etc/systemd/network/wg0.network` is run before `networkctl reload`. even if `networkctl delete wg0` is run, the old config file is still used during reload. strangely, both the original and drop-in files are re-read, but the changes are not applied.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #15521.

### CODE REVIEW

File: src/shared/conf-parser.h [hunk 1]
```diff
@@ -100,6 +100,7 @@ int config_parse_many_nulstr(
                 void *userdata);
 
 int config_parse_many(
+                char ***ret_dropins,        /* possibly NULL */
                 const char *conf_file,      /* possibly NULL */
                 const char* const* conf_file_dirs,
                 const char *dropin_dirname,
```
Comment (keszybz) [inline src/shared/conf-parser.h]: Looks nice, except for one nitpick: I'd prefer to have the output parameter last. I know this is a bit of bike-shedding, and if you disagree I'm fine with leaving this form, but in general the coding style is to have output parameters last. The only exception is "constructor" functions, which doesn't apply to this case.

File: src/network/networkd-network.c [hunk 2]
```diff
@@ -542,6 +543,17 @@ int network_load_one(Manager *manager, OrderedHashmap **networks, const char *fi
                 return -errno;
```
Comment (keszybz) [inline src/network/networkd-network.c]: In other places where we do stat for this purpose, we set the mtime to 0 on error. I think that's nice than propagating the error. We'll just maybe reload the config unnecessarily later.

File: src/shared/conf-parser.h (hunk 1 above)
Comment (yuwata) [reply src/shared/conf-parser.h]: OK. Done.

File: src/network/networkd-network.c (hunk 2 above)
Comment (yuwata) [reply src/network/networkd-network.c]: OK. Updated.

**Comment by yuwata** [comment]
@keszybz Thank you for the review. I've updated the commits with your suggestions. PTAL.

File: src/network/networkd-network.c [hunk 3]
```diff
@@ -541,6 +542,17 @@ int network_load_one(Manager *manager, OrderedHashmap **networks, const char *fi
         if (stat(filename, &stats) >= 0)
                 network->timestamp = timespec_load(&stats.st_mtim);
 
+        char **f;
+        STRV_FOREACH(f, dropins) {
+                usec_t t;
+
+                if (stat(*f, &stats) < 0)
+                        continue;
```
Comment (keszybz) [inline src/network/networkd-network.c]: Hmm, that's not correct either. I think `network->timestamp` should be reset to 0 here and `break`.

Comment (yuwata) [reply src/network/networkd-network.c]: I see. Updated.

**Comment by keszybz** [comment]
LGTM>

**Pull request merged**